diff --git a/node_modules/notion-rehype-k/dist/handlers/callout.js b/node_modules/notion-rehype-k/dist/handlers/callout.js
index 33d0c84..7ee8cee 100644
--- a/node_modules/notion-rehype-k/dist/handlers/callout.js
+++ b/node_modules/notion-rehype-k/dist/handlers/callout.js
@@ -6,12 +6,13 @@ const handler = (context, block) => {
     const data = block[BlockType.callout];
     const blockClass = notionPrefixFactory(context)(BlockType.callout);
     const calloutIcon = h('div', { className: [`${blockClass}-icon`] }, []);
-    if (data.icon.type === 'emoji') {
-        calloutIcon.children.push(h('text', data.icon.emoji));
-    }
-    else {
-        const { url, attr: fileAttr } = getNotionFileUrlAndAttr(context, data.icon);
-        calloutIcon.children.push(h('img', { src: url, ...fileAttr }, []));
+    if (data.icon) {
+        if (data.icon.type === 'emoji') {
+            calloutIcon.children.push(h('text', data.icon.emoji));
+        } else {
+            const { url, attr: fileAttr } = getNotionFileUrlAndAttr(context, data.icon);
+            calloutIcon.children.push(h('img', { src: url, ...fileAttr }, []));
+        }
     }
     const calloutContent = h('div', { className: [`${blockClass}-content`] }, []);
     addTasksToAddDirectChildren(context, calloutContent, data.children);
diff --git a/node_modules/notion-rehype-k/dist/handlers/code.js b/node_modules/notion-rehype-k/dist/handlers/code.js
index d4d7962..441d425 100644
--- a/node_modules/notion-rehype-k/dist/handlers/code.js
+++ b/node_modules/notion-rehype-k/dist/handlers/code.js
@@ -11,7 +11,7 @@ const handler = (context, block) => {
         return pre;
     }
     const blockClass = notionPrefixFactory(context)(BlockType.code);
-    const hast = h('section', { className: [blockClass] }, [pre]);
+    const hast = h('div', { className: [blockClass] }, [pre]);
     addCaptionToHast(context, hast, data.caption);
     return hast;
 };
diff --git a/node_modules/notion-rehype-k/dist/handlers/embed.js b/node_modules/notion-rehype-k/dist/handlers/embed.js
index 491cbba..b5d67b9 100644
--- a/node_modules/notion-rehype-k/dist/handlers/embed.js
+++ b/node_modules/notion-rehype-k/dist/handlers/embed.js
@@ -3,7 +3,22 @@ import { BlockType } from '../types.js';
 const handler = (context, block) => {
     const data = block[BlockType.embed];
     const blockClass = notionPrefixFactory(context)(BlockType.embed);
-    const hast = h('iframe', { src: data.url, className: [blockClass] }, []);
+
+    let hast;
+
+    if (data.url.includes("x.com")) {
+        const url = data.url.replace("x.com", "twitter.com");
+
+        hast = h("div", { className: [blockClass]}, [
+            h("blockquote", { className: ["twitter-tweet w-full"], dataTheme: "dark" }, [
+                h("a", { href: url }, [])
+            ]),
+            h("script", { async: true, src: "https://platform.twitter.com/widgets.js", charset: "utf-8" }, [])
+        ]);
+    } else {
+        hast = h('iframe', { src: data.url, className: [blockClass] }, []);
+    }
+    
     return hast;
 };
 export default handler;
diff --git a/node_modules/notion-rehype-k/dist/handlers/video.js b/node_modules/notion-rehype-k/dist/handlers/video.js
index 35704a5..a814503 100644
--- a/node_modules/notion-rehype-k/dist/handlers/video.js
+++ b/node_modules/notion-rehype-k/dist/handlers/video.js
@@ -3,9 +3,20 @@ import { h, getNotionFileUrlAndAttr, notionPrefixFactory } from '../utils.js';
 import { BlockType } from '../types.js';
 const handler = (context, block) => {
     const data = block[BlockType.video];
-    const { url, attr: fileAttr } = getNotionFileUrlAndAttr(context, data);
+    const videoUrl = new URL(data.external.url);
+    
+    const videoId = videoUrl.searchParams.get("v");
     const blockClass = notionPrefixFactory(context)(BlockType.video);
-    const hast = h('div', { className: [blockClass] }, [h('video', { src: url, ...fileAttr }, [])]);
+    const hast = 
+        h(
+            'div',
+            { className: [blockClass] }, 
+            [
+                h('button', {  dataVideoId: videoId, ariaLabel: "Play video" }, [
+                    h('img', { src: `https://img.youtube.com/vi/${videoId}/maxresdefault.jpg` }, [])
+                ])
+            ]
+        );
     addCaptionToHast(context, hast, data.caption);
     return hast;
 };
